### PR TITLE
Fix circular deps

### LINKED ISSUE #5087: ChipGroup throwing TypeError
**Describe the issue. What is the expected and unexpected behavior?**
@Hyperkid123 and @karelhala report that Insights is throwing the following error when [using ChipGroup here](https://github.com/RedHatInsights/insights-chrome/blob/master/src/js/App/GlobalFilter/GlobalFilter.js#L134):

Screenshot from 2020-10-14 11-07-56

The issue persists after upgrading to latest PF-react release 4.75.2, per @karelhala's investigation:

> Turns out there is circular dependency, the functions get bundled correctly, everything is running as it should, however upon importing any helper function it directly displays `undefined` instead of the implementation. When I imported the whole helpers it was undefined, but after a while the object changed. This is usually a problem indicating circular dependencies as one component imports another which imports the entire helpers file.

> I'd suspect that either `ApplicationLauncherSeparator` or `Divider` imports whole helpers or another component that imports it. Can you take a look at these components if that's really the case?

### PR DESCRIPTION
**Pull request by redallen**
**What**: Possibly fixes build issue in https://github.com/redhatInsights/insights-chrome per @karelhala 's suggestion.
Closes #5087 

- Start bundling react-table and react-datetime as UMD in addition to react-core
  - Use common `rollup.base.js`
- Treat circular deps as errors
- Fix circular dep errors in react-core, react-table, and react-charts (react-charts will always have circular deps due to d3 having internal circular deps, so not adding `build:umd` to it for now)

**Additional issues**: Circular deps are bad practice, so cleaning this up is good even if it doesn't solve the CRC build problem.

### CODE REVIEW

File: packages/react-table/src/components/Table/TableContext.ts [hunk 1]
```diff
@@ -0,0 +1,232 @@
+import * as React from 'react';
+import {
+  DropdownDirection,
+  DropdownPosition
+} from '@patternfly/react-core/dist/js/components/Dropdown/dropdownConstants';
+import { DropdownItemProps } from '@patternfly/react-core/dist/js/components/Dropdown/DropdownItem';
+import { formatterValueType, ColumnType, RowType, RowKeyType, ColumnsType } from './base';
+
+export type RowEditType = 'save' | 'cancel' | 'edit';
```
[the remark this answers is not in the document]
Comment (redallen) [reply packages/react-table/src/components/Table/TableContext.ts:9]: There was a circular dependency between Table -> ... -> Transformers -> Selectable (which needed `RowSelectVariant`) -> Table. My solution was to move `RowSelectVariant` along with `TableContext` (which also had a circular dep) outside the root `Table.tsx` file.

[the remark this answers is not in the document]
Comment (jschuler) [reply packages/react-table/src/components/Table/TableContext.ts:9]: how about moving all the types into `./base/types.tsx`

[the remark this answers is not in the document]
Comment (redallen) [reply packages/react-table/src/components/Table/TableContext.ts]: I reverted most changes and only moved the `SortByDirection` and `RowSelectVariant` enums into their respective column files. Please rereview!

**Review by jschuler — APPROVED** [review]

File: packages/react-charts/src/components/ChartCursorTooltip/ChartCursorTooltip.tsx [hunk 2]
```diff
@@ -12,8 +12,9 @@ import {
 import { VictoryTooltip } from 'victory-tooltip';
 import { ChartLabel } from '../ChartLabel';
 import { ChartThemeDefinition } from '../ChartTheme';
-import { ChartTooltip, ChartTooltipProps } from '../ChartTooltip';
-import { getTheme, getCursorTooltipCenterOffset, getCursorTooltipPoniterOrientation } from '../ChartUtils';
+import { ChartTooltip, ChartTooltipProps } from '../ChartTooltip/ChartTooltip';
+import { getTheme } from '../ChartUtils/chart-theme';
+import { getCursorTooltipCenterOffset, getCursorTooltipPoniterOrientation } from '../ChartUtils/chart-tooltip';
```
Comment (dlabrecq) [inline packages/react-charts/src/components/ChartCursorTooltip/ChartCursorTooltip.tsx:17]: Can you please explain these changes? I'm afraid these types of things may be reverted without some comments as to why the module index cannot be used.

**Review by evwilkin — APPROVED** [review]
LGTM

**Review by dlabrecq — CHANGES_REQUESTED** [review]

File: packages/react-charts/src/components/ChartCursorTooltip/ChartCursorTooltip.tsx (hunk 2 above)
Comment (redallen) [reply packages/react-charts/src/components/ChartCursorTooltip/ChartCursorTooltip.tsx:17]: I'll revert charts changes, I just fixed them as a best-effort to adhere to good coding practices. These are the errors it fixed if we were to bundle charts as UMD:
```sh
(!) Circular dependency: dist/esm/components/ChartUtils/index.js ->
  dist/esm/components/ChartUtils/chart-container.js ->
  dist/esm/components/ChartCursorTooltip/index.js ->
  dist/esm/components/ChartCursorTooltip/ChartCursorTooltip.js ->
  dist/esm/components/ChartTooltip/index.js ->
  dist/esm/components/ChartTooltip/ChartTooltip.js ->
  dist/esm/components/ChartUtils/index.js
(!) Circular dependency: dist/esm/components/ChartUtils/index.js ->
  dist/esm/components/ChartUtils/chart-container.js ->
  dist/esm/components/ChartCursorTooltip/index.js ->
  dist/esm/components/ChartCursorTooltip/ChartCursorTooltip.js ->
  dist/esm/components/ChartUtils/index.js
```

**Review by evwilkin — APPROVED** [review]
LGTM

**Review by dlabrecq — APPROVED** [review]

**Review by jschuler — APPROVED** [review]

**Pull request merged**